### PR TITLE
implement comment at-mentions

### PR DESCRIPTION
**Pull request by caolanm**
* Resolves: #1529 
* Target version: master

### CODE REVIEW

File: browser/src/canvas/sections/CommentSection.ts [hunk 1]
```diff
@@ -508,7 +508,7 @@ export class Comment extends CanvasSectionObject {
 		// Original unlinked text
 		this.sectionProperties.contentText.origText = this.sectionProperties.data.text ? this.sectionProperties.data.text: '';
 		this.sectionProperties.nodeModifyText.textContent = this.sectionProperties.data.text ? this.sectionProperties.data.text: '';
-		this.sectionProperties.nodeModifyText.value = this.sectionProperties.data.text ? this.sectionProperties.data.text: '';
+		this.sectionProperties.nodeModifyText.textContent = this.sectionProperties.data.text ? this.sectionProperties.data.text: '';
```
Comment (caolanm) [inline browser/src/canvas/sections/CommentSection.ts:506]: @Rash419 we end up with the same line here twice, so presumably we don't need one of those?

File: browser/src/control/AutoCompletePopup.ts [hunk 2]
```diff
@@ -136,16 +118,45 @@ abstract class AutoCompletePopup {
 		});
 	}
 
+	getCursorPosition(): Point {
+		const commentSection = app.sectionContainer.getSectionWithName(
+			L.CSections.CommentList.name,
+		);
+
+		if (commentSection?.sectionProperties?.selectedComment?.isEdit()) {
```
Comment (caolanm) [inline browser/src/control/AutoCompletePopup.ts:126]: (bizarrely) the "selectedComment" appears to actually be the *root* comment of the current comment. So if we insert a comment, and then reply to the comment and "ok" and then edit the reply and use an at-mention. The selectedComment is the root comment, which is not actually being edit and the at-mention popup appears in the main document

File: browser/src/control/AutoCompletePopup.ts [hunk 3]
```diff
@@ -65,11 +65,12 @@ abstract class AutoCompletePopup {
 	abstract onAdd(): void;
 
 	closePopup(): void {
-		var popupExists = L.DomUtil.get(this.popupId);
+		var popupExists = L.DomUtil.get(this.popupId + 'List');
```
Comment (caolanm) [inline browser/src/control/AutoCompletePopup.ts:70]: @Rash419 After this change, if I type @ followed by something that doesn't mater, e.g. aaaa, then the 'no search results found' popup doesn't disappear on pressing esc

**Review by eszkadev — APPROVED** [review]
looks fine

**Pull request merged**